protoc-gen-swagger/options/openapiv2.proto

```diff
@@ -84,8 +84,9 @@ message Response {
   Schema schema = 2;
   // field 3 is reserved for 'headers'.
   reserved 3;
-  // field 3 is reserved for 'example'.
-  reserved 4;
+  // `Examples` gives per-mimetype response examples.
+  // See: https://github.com/OAI/OpenAPI-Specification/blob/3.0.0/versions/2.0.md#example-object
+  map<string, string> examples = 4;
```

Comment:
I had a brief discussion with the other maintainers about this again and we're curious if it's possible to do this with a `map<string, any.Any>`. Have you tried it? What does a literal look like? Don't worry about spending time on it, just wanted to explore the possibility.

Reply:
I'm not sure `google.protobuf.Any` makes sense semantically here: https://github.com/protocolbuffers/protobuf/blob/4cf5bfee9546101d98754d23ff378ff718ba8438/src/google/protobuf/any.proto#L122

The `string type_url` value interpretation appears grpc-specific and unlikely to be useful in a Swagger context. In any case, its intent would appear to duplicate the `schema` property in the containing `Response` message here (and that `schema` field, in contrast, *is* meaningfully interpreted by Swagger tooling).

Examining the existing uses of `google.protobuf.Any` in these annotations (in particular, `Schma.example` https://github.com/grpc-ecosystem/grpc-gateway/blob/f0d0de466d4242a713e95d64685c12064495284a/protoc-gen-swagger/options/openapiv2.proto#L152 ), it would appear that the example literals omit the `type_url` field (questionable) and then use the fact that the `bytes value` field can be specified with a string literal to give a JSON string example: https://github.com/grpc-ecosystem/grpc-gateway/blob/f0d0de466d4242a713e95d64685c12064495284a/examples/proto/examplepb/a_bit_of_everything.proto#L196

So the `google.protobuf.Any` application looks like a semantically-questionable, more-verbose way to achieve the effect of the current `map<string, string>`.

On the other hand, a pretty good argument could be made for `map<string, google.protobuf.Value>` here, similar to how it is used in the various `extensions` fields (e.g. https://github.com/grpc-ecosystem/grpc-gateway/blob/f0d0de466d4242a713e95d64685c12064495284a/protoc-gen-swagger/options/openapiv2.proto#L342 ) . The main argument against it is the verbosity relative to a straight JSON string representation (see e.g. https://github.com/grpc-ecosystem/grpc-gateway/blob/f0d0de466d4242a713e95d64685c12064495284a/examples/proto/examplepb/a_bit_of_everything.proto#L66 for a simple two-field case). The verbosity is particularly unfortunate because in the 99% case here you are writing an example that conforms to an already-defined schema definition, meaning the explicit structuring is not particularly helpful.

Reply:
Thanks, that makes sense